addons/mail/models/mail_channel.py

```diff
@@ -63,13 +68,14 @@ def default_get(self, fields):
         'mail.channel.partner', 'channel_id', string='Last Seen',
         groups='base.group_user')
     is_member = fields.Boolean('Is Member', compute='_compute_is_member', compute_sudo=True)
+    member_count = fields.Integer(string="Member Count", compute='_compute_member_count', help="Excluding guests from count.")
     group_ids = fields.Many2many(
         'res.groups', string='Auto Subscription',
         help="Members of those groups will automatically added as followers. "
              "Note that they will be able to manage their subscription manually "
              "if necessary.")
     # access
-    uuid = fields.Char('UUID', size=50, index=True, default=lambda self: str(uuid4()), copy=False)
+    uuid = fields.Char('UUID', size=50, index=True, default=lambda self: self._generate_random_token(8), copy=False)
```

Comment:
why 8?? that's much less entropy than before (~ 122bit vs 45!) and it also fails to account for hard-to-distinguish characters (e.g. 1 vs I vs l vs 0 vs o vs O), if readability and length were factors.

You need 21 chars to keep the entropy (or 22 if you use a hardcoded list of easy-to-read chars).

Reply:
The goal is to have a rather short URL to share. It's 8 char for a functional reason because that's how many are on share link in Discord.

Good point to exclude hard-to-distinguish but then it's even worse in terms of security...

Ideally I'd like that we generate the uuid only when we actually need to share it, but some other parts of the code still rely on it being always set for now...

Do you have any suggestion with this in mind?
Do we actually need as much entropy as before, or can we settle for a middle ground?

Maybe at least log something in case of wrong value in the invite route so that fail2ban can be set to prevent the most simple brute force?

Note: if this is not negotiable I'll put 22, I don't want to block the PR for this.

Reply:
Renaming is too much work (and risk of breaking other stuff) for the time I have available now, but ok to improve after v15. This PR opens a lot of clean up opportunities also in portal chatter and livechat.

Reply:
Made it to be 10 with the provided list.